FlowConcatSpec

https://jenkins.akka.io:8498/job/akka-multi-node/1270/consoleFull

```
[info] - must work with one nonempty and one delayed failed publisher *** FAILED *** (25 milliseconds)
[info]   java.lang.AssertionError: assertion failed: fishForMessage(OnNext(1) or akka.stream.testkit.BaseTwoStreamsSetup$$anon$1) found unexpected message OnNext(2)
[info]   at scala.Predef$.assert(Predef.scala:165)
[info]   at akka.testkit.TestKitBase$class.recv$1(TestKit.scala:399)
[info]   at akka.testkit.TestKitBase$class.fishForMessage(TestKit.scala:402)
[info]   at akka.testkit.TestKit.fishForMessage(TestKit.scala:735)
[info]   at akka.stream.testkit.TestSubscriber$ManualProbe.expectNextOrError(StreamTestKit.scala:465)
[info]   at akka.stream.scaladsl.FlowConcatSpec$$anonfun$1$$anonfun$apply$mcV$sp$8$$anonfun$apply$mcV$sp$21$$anonfun$7.apply(FlowConcatSpec.scala:114)
[info]   at akka.stream.scaladsl.FlowConcatSpec$$anonfun$1$$anonfun$apply$mcV$sp$8$$anonfun$apply$mcV$sp$21$$anonfun$7.apply(FlowConcatSpec.scala:113)
[info]   at scala.collection.TraversableOnce$$anonfun$foldLeft$1.apply(TraversableOnce.scala:155)
[info]   at scala.collection.TraversableOnce$$anonfun$foldLeft$1.apply(TraversableOnce.scala:155)
[info]   at scala.collection.immutable.Range.foreach(Range.scala:166)
[info]   at scala.collection.TraversableOnce$class.foldLeft(TraversableOnce.scala:155)
[info]   at scala.collection.AbstractTraversable.foldLeft(Traversable.scala:104)
[info]   at 
```

Maybe caused by the prefetch logic?

--- a/akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowConcatSpec.scala
+++ b/akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowConcatSpec.scala
@@ -98,7 +98,7 @@ class FlowConcatSpec extends BaseTwoStreamsSetup {
       subscriber.expectSubscription().request(5)
 
       val errorSignalled = (1 to 4).foldLeft(false)((errorSignalled, e) ⇒
-        if (!errorSignalled) subscriber.expectNextOrError(1, TestException).isLeft else true)
+        if (!errorSignalled) subscriber.expectNextOrError(e, TestException).isLeft else true)
       if (!errorSignalled) subscriber.expectSubscriptionAndError(TestException)
     }
 
@@ -112,7 +112,7 @@ class FlowConcatSpec extends BaseTwoStreamsSetup {
       subscriber.expectSubscription().request(5)
 
       val errorSignalled = (1 to 4).foldLeft(false)((errorSignalled, e) ⇒
-        if (!errorSignalled) subscriber.expectNextOrError(1, TestException).isLeft else true)
+        if (!errorSignalled) subscriber.expectNextOrError(e, TestException).isLeft else true)
       if (!errorSignalled) subscriber.expectSubscriptionAndError(TestException)
     }
 

